diff --git mojo/core/channel_linux.cc mojo/core/channel_linux.cc
index 20333a7d19..9464ea58af 100644
--- mojo/core/channel_linux.cc
+++ mojo/core/channel_linux.cc
@@ -48,6 +48,25 @@
 #define EFD_ZERO_ON_WAKE O_NOFOLLOW
 #endif
 
+#ifndef F_LINUX_SPECIFIC_BASE
+#define F_LINUX_SPECIFIC_BASE 1024
+#endif
+#ifndef F_SEAL_SEAL
+#define F_SEAL_SEAL 0x0001
+#endif
+#ifndef F_SEAL_SHRINK
+#define F_SEAL_SHRINK 0x0002
+#endif
+#ifndef F_SEAL_GROW
+#define F_SEAL_GROW 0x0004
+#endif
+#ifndef F_ADD_SEALS
+#define F_ADD_SEALS (F_LINUX_SPECIFIC_BASE + 9)
+#endif
+#ifndef F_GET_SEALS
+#define F_GET_SEALS (F_LINUX_SPECIFIC_BASE + 10)
+#endif
+
 namespace mojo {
 namespace core {
 
diff --git sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
index a8f860fcf0..e007d1f505 100644
--- sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
+++ sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
@@ -72,6 +72,25 @@
 #define PROT_BTI 0x10
 #endif
 
+#ifndef F_LINUX_SPECIFIC_BASE
+#define F_LINUX_SPECIFIC_BASE 1024
+#endif
+#ifndef F_SEAL_SEAL
+#define F_SEAL_SEAL 0x0001
+#endif
+#ifndef F_SEAL_SHRINK
+#define F_SEAL_SHRINK 0x0002
+#endif
+#ifndef F_SEAL_GROW
+#define F_SEAL_GROW 0x0004
+#endif
+#ifndef F_ADD_SEALS
+#define F_ADD_SEALS (F_LINUX_SPECIFIC_BASE + 9)
+#endif
+#ifndef F_GET_SEALS
+#define F_GET_SEALS (F_LINUX_SPECIFIC_BASE + 10)
+#endif
+
 namespace {
 
 inline bool IsArchitectureX86_64() {
diff --git sandbox/policy/linux/bpf_gpu_policy_linux.cc sandbox/policy/linux/bpf_gpu_policy_linux.cc
index b19668b880..b56aa79f78 100644
--- sandbox/policy/linux/bpf_gpu_policy_linux.cc
+++ sandbox/policy/linux/bpf_gpu_policy_linux.cc
@@ -23,6 +23,25 @@
 #include "sandbox/policy/linux/sandbox_linux.h"
 #include "sandbox/policy/linux/sandbox_seccomp_bpf_linux.h"
 
+#ifndef F_LINUX_SPECIFIC_BASE
+#define F_LINUX_SPECIFIC_BASE 1024
+#endif
+#ifndef F_SEAL_SEAL
+#define F_SEAL_SEAL 0x0001
+#endif
+#ifndef F_SEAL_SHRINK
+#define F_SEAL_SHRINK 0x0002
+#endif
+#ifndef F_SEAL_GROW
+#define F_SEAL_GROW 0x0004
+#endif
+#ifndef F_ADD_SEALS
+#define F_ADD_SEALS (F_LINUX_SPECIFIC_BASE + 9)
+#endif
+#ifndef F_GET_SEALS
+#define F_GET_SEALS (F_LINUX_SPECIFIC_BASE + 10)
+#endif
+
 using sandbox::bpf_dsl::Allow;
 using sandbox::bpf_dsl::Arg;
 using sandbox::bpf_dsl::Error;
diff --git ui/events/ozone/evdev/event_device_info.cc ui/events/ozone/evdev/event_device_info.cc
index 08b09f3b99..9c379060f3 100644
--- ui/events/ozone/evdev/event_device_info.cc
+++ ui/events/ozone/evdev/event_device_info.cc
@@ -19,6 +19,10 @@
 #define EVIOCGMTSLOTS(len) _IOC(_IOC_READ, 'E', 0x0a, len)
 #endif
 
+#ifndef SW_PEN_INSERTED
+#define SW_PEN_INSERTED                0x0f  /* set = pen inserted */
+#endif
+
 namespace ui {
 
 namespace {
diff --git ui/events/ozone/evdev/event_converter_evdev_impl.cc ui/events/ozone/evdev/event_converter_evdev_impl.cc
index 6b879e29d8..38b48aa7e4 100644
--- ui/events/ozone/evdev/event_converter_evdev_impl.cc
+++ ui/events/ozone/evdev/event_converter_evdev_impl.cc
@@ -17,6 +17,10 @@
 #include "ui/events/ozone/evdev/device_event_dispatcher_evdev.h"
 #include "ui/events/ozone/evdev/event_device_util.h"
 
+#ifndef SW_PEN_INSERTED
+#define SW_PEN_INSERTED                0x0f  /* set = pen inserted */
+#endif
+
 namespace ui {
 
 namespace {
